module-2/backend/week_one.md

```diff
@@ -5,30 +5,112 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. List the five common HTTP verbs and what the purpose is of each verb.
+
+  post, get, put, delete, options
+
 2. What is Sinatra?
+
+  Sinatra is a MVC based web framework
+
 4. What is MVC?
-5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+
+  Models, Views, Controllers
+
+5. Why do we follow conventions when creating our actions/path names in our Sinatra routes? 
+
+  they accomplish a simplified way to read the path names and make connections between desired actions
+
 6. What types of variables are accessible in our view templates without explicitly passing them?
+
+  the variables can represent the methods in the models, routes 
+
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+  @count = 1
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+
+  ```ruby
+  get '/horses' do
+  @count = 1
+  @name = Mr. Ed
+    erb :index
+  end
+  ```
+
 9. What's the purpose of ERB?
+
+  embedded ruby: is a feature of ruby that enables combining plain text with Ruby code, making generating text easy
+
 10. Why do I need a development AND test database?
+
+  as a precaution against accidents and it provides the ability to recognise which system you're on so you preform actions     on the correct system
+
 11. What's responsive design?
+
+  Responsive Web Design (RWD) is an approach of crafting sites that provide an optimal viewing experience
+
 12. What is CRUD and why is it important?
+
+  C = Create
+  R = Read
+  U = Update
+  D = Delete
+  
+  it is important because it facilitates user viewing, searching and changing information using forms
+
+
 13. What does HTTP stand for? 
+
+  Hyper Text Transfer Protocol
+    the foundation of data communication for the World Wide Web
+    application protocol for collaborative hypermedia information systems
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+  examples: <%= @horse.id %> shows whatever is after the = 
+            <% @horse.id %> just runs code
+
 15. What's an ORM?
+
+  object relational mappers: converts data between incompatible type systems in object oriented programming languages
+
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+
+  ActiveRecord
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+get '/'
+get '/whatever'
+get '/whatever/new'
+get '/whatever/:id'
+get '/whatever/:id/edit'
+put '/whatever/:id'
+delete '/whatever/:id'

```

Comment:
Looks like we're missing a POST to '/restaurants' (GET '/' isn't part of the CRUD functionality). Also, you need to describe what each of these does!